tests/functional/hooks/test_model_hooks.py

```diff
@@ -263,6 +265,36 @@ def test_hooks_on_seeds(self, project):
         assert len(res) == 1, "Expected exactly one item"
 
 
+class TestHooksRefsOnSeeds:
+    """
+    This should not succeed, and raise an explicit error
+    https://github.com/dbt-labs/dbt-core/issues/6806
+    """
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"example_seed.csv": seeds__example_seed_csv}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"schema.yml": properties__seed_models, "post.sql": models__post}
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "seeds": {
+                "post-hook": [
+                    "select * from {{ ref('post') }}",
```

Comment:
I want to make sure I understand this completely. Does this mean that we can't have a model depending on the seed either? Or does it just prevent the seed itself from depending on another object? I'm asking because I've written several integration tests that load a seed table, then take various models based off of that seed table to test functionality. My guess is other folks may have done the same.

Reply:
Yup, @gshank is right here. Seeds are root nodes in the DAG — they can have children (models), but they cannot have parents. In previous versions, it was theoretically possible to force a seed to take a dependency on another node, by attaching a hook that included a `ref()` call. This is bad! We shouldn't let people do it.

Reply:
Got it. Thanks @gshank and @jtcohen6!